backend/de.metas.handlingunits.base/src/main/java/de/metas/picking/service/impl/HU2PackingItemsAllocator.java

```diff
@@ -491,29 +511,104 @@ private void pickFromVHU(@NonNull final I_M_HU pickFromVHU, @NonNull final Packi
 
 		final I_M_ShipmentSchedule shipmentSchedule = getShipmentScheduleById(packedPart.getShipmentScheduleId());
 
+		final BigDecimal currentQtyToDeliver = shipmentSchedule.getQtyToDeliver();
+		final boolean isOverDelivery = currentQtyToDeliver.compareTo(qtyPacked.toBigDecimal()) < 0;
+
 		// Check over-delivery
-		if (!allowOverDelivery)
+		if (!allowOverDelivery && isOverDelivery && TakeWholeHUEnum.NONE.equals(takeWholeHU.getCode()))
 		{
-			final BigDecimal currentQtyToDeliver = shipmentSchedule.getQtyToDeliver();
-			if (currentQtyToDeliver.compareTo(qtyPacked.toBigDecimal()) < 0)
-			{
 				throw new AdempiereException("@" + PickingConfigRepository.MSG_WEBUI_Picking_OverdeliveryNotAllowed + "@")
 						.setParameter("shipmentSchedule's QtyToDeliver", currentQtyToDeliver)
 						.setParameter("qtyPacked to be Delivered", qtyPacked);
-			}
 		}
 
-		final StockQtyAndUOMQty stockQtyAndUomQty = CatchWeightHelper.extractQtys(_huContext, getProductId(), qtyPacked, pickFromVHU);
+		if (allowOverDelivery && isOverDelivery && TakeWholeHUEnum.SPLIT.getCode().equals(takeWholeHU.getCode()))
+		{
+			splitCurrentHU(pickFromVHU, packedPart);
+		}
+
+		else
+		{
+			final StockQtyAndUOMQty stockQtyAndUomQty = CatchWeightHelper.extractQtys(_huContext, getProductId(), qtyPacked, pickFromVHU);
+
+			// "Back" allocate the qtyPicked from VHU to given shipment schedule
+			final boolean anonymousHuPickedOnTheFly = false;
+			huShipmentScheduleBL.addQtyPickedAndUpdateHU(shipmentSchedule, stockQtyAndUomQty, pickFromVHU, _huContext, anonymousHuPickedOnTheFly);
+
+			// Transfer the qtyPicked from vhu to our target HU (if any)
+			packFromVHUToDestination(pickFromVHU, packedPart);
+
+			// Adjust remaining Qty to be packed
+			subtractFromQtyToPackRemaining(qtyPacked);
+		}
+	}
+
+	private void splitCurrentHU(@NonNull final I_M_HU pickFromVHU, @NonNull final PackingItemPart packedPart)
+	{
+		final Quantity qtyPacked = packedPart.getQty();
+		final I_M_ShipmentSchedule shipmentSchedule = getShipmentScheduleById(packedPart.getShipmentScheduleId());
+		final BigDecimal currentQtyToDeliver = shipmentSchedule.getQtyToDeliver();
+
+		final Quantity qtyCU = Quantity.of(currentQtyToDeliver, packedPart.getQty().getUOM());
+
+		// e.g. if we have one 1000PCE-HU, but we need to issue just 10PCE, then we need to split those 10PCE from the big HU
+
+		final ProductId productId = packedPart.getProductId();
+		final I_M_HU huToIssue = CollectionUtils.singleElement(HUTransformService.newInstance()
```

Comment:
what to issue?
aren't we talking about HUs which were RECEIVED from manufacturing order?

I saw that u used "issue" all over but this feature is not about issueing. Can u pls check & correct?